src/com/palmergames/bukkit/towny/object/Town.java

```diff
@@ -254,7 +259,12 @@ public boolean hasResident(Resident resident) {
 
 	public boolean hasAssistant(Resident resident) {
 
-		return getAssistants().contains(resident);
+		return hasRank(resident, "assistant");
```

Comment:
this should probably just `return resident.hasTownRank("assistant");` and the hasRank() method below can be scrapped. Eventually I want all of the assistant stuff removed from being hardcoded, including the saving/loading assistants in the DB.

Reply:
Is `Resident.hasTownRank()` new?

Reply:
No it's fairly old.

Reply:
Oh yeah, I used it in `Town.findNewMayor(String rank)`.  I'm not sure why I set up Town.hasRank(Resident resident, String rank)` that way except that that was the way `Town.hasAssistant(Resident resident)` originally worked.

Reply:
Commits ea3df36b146343b759d970e672d52a33ea9c72ad and fb20ac24c5f90fcc964700766eff98cf54398eaa resolve this.